fix(Types): make null and any interoperability work

adnrs96 opened the pull request:
We make sure that `null` could be implicitly converted to `any`
so that it can be passed around in function arguments accepting
`any` as parameter.
Also I added another commit which prevents `null` being used in indexs. I believe we need this.
It stops things like
```
a = [1, 2]
a[null]
```

Fixes: #1276

wilzbach commented on storyscript/compiler/semantics/symbols/Symbols.py:
```diff
@@ -134,6 +134,11 @@ def index(self, tree, name, type_, kind):
+                    'type_index_incompatible',
+                    left=cur_type,
+                    name=name,
+                    right=type_)
```
Uhm we should better add this to `Types.py` to avoid special cases when `.index` called from somewhere else.

wilzbach replied:
Also, I think the check in general shouldn't be inserted as we need to treat `null` like `any`.
For example:

```
a = null
b = [] to List[any]
a = 1
c = b[a]
```

and a slight variation of `null_map.story`:

```
a = null
b = {} to Map[any,any]
a = "foo"
c = b[a]
```

adnrs96 commented:
@wilzbach I added the last commit in this PR after taking your comment into account. After thinking a bit more about it, I removed the check and also changed the test for `null_map.story` from failing to pass because if we are going to allow NullTypes in indexs as Anytypes are allowed (any in index makes sense for multiple data type keys), we can't reject any NullType variable and it must the be the job of the runtime to figure the value of this variable is `null` and raise a runtime Error.

To be honest, I don't like it. I would have loved to error out in compilation if I could tell the value at the moment is `null` but I can't at the moment. We should thinking about storing values wrapped inside the types?

wilzbach commented:
> if we are going to allow NullTypes in indexs as Anytypes are allowed (any in index makes sense for multiple data type keys), we can't reject any NullType variable and it must the be the job of the runtime to figure the value of this variable is null and raise a runtime Error.

Yes.

> To be honest, I don't like it. I would have loved to error out in compilation if I could tell the value at the moment is null but I can't at the moment. We should thinking about storing values wrapped inside the types?

Please feel free to create an issue (+ one for any), but for now that's the job of the runtime.

adnrs96 commented:
Created #1286, #1287
